presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1720,6 +1757,24 @@ else if (criteria instanceof JoinOn) {
                     }
                 }
 
+                if (!checkIfRelationUsedInExpressionJointlyWithOtherRelation(expression, node.getRight())) {
+                    String rightTableName = "";
+                    if (node.getRight() instanceof Table) {
```

Comment:
Also I think we should do these checks after the resolution. for example:

select t1 join t2 on x1=y1 join t3 on x2=y2

where x2 is from t1 and y2 is from t2 unambiguously won't be caught by this syntactic way of checking

Reply:
It is better to support this case in a separate task.